### ISSUE TITLE
Understrap + Bootstrap + Vue.js

### ISSUE THREAD

**Issue by raisonon** [understrap/understrap#537]
I'm looking at the future of my applications with Understrap and need to upgrade from jQuery. In particular, I'm looking at using the Vue.js framework/library.

I've spotted this project: https://bootstrap-vue.js.org/

I have some free days put aside to learn and may attempt a fork of understrap which uses this. I was curious to whether this has been considered by others here and general feedback.

Cheers!

**Comment by vule840** [comment]
+

**Comment by vule840** [comment]
https://github.com/bedakb/vuewp

**Comment by Thomas-A-Reinert** [comment]
Please shed some light: Afaik there´s already REACT integrated into WP-Core.

- Is there any (reasonable) reason to bloat the system with another JS-Framework? 
- Would it help any other user?

**Comment by raisonon** [comment]
Hi Thomas, great questions. 

1. There will be **no bloat** as the proposal here is a swap of jQuery with vue.js. Not adding alongside.
2. **Would it help others? Yes!** I have hit the limitations of vanilla JS and jQuery and expect others using understrap to build bespoke sites will too. Using a reactive JS framework is a business requirement I have and I am confident this will be even more the case for all users in the coming year.

**A bit more detail:**
1. Why vue.js and not react? Both are great. React is only currently used in the wp-admin with Gutenberg but would require a major rewrite of Understrap templates whereas vue.js can just be swapped in to replace jQuery. The PHP/HTML of the templates can exist as they are and we can swap in vue.js.

2. Bootstrap already has a fork working with vue.js. Currently, Understrap is using jQuery as Bootstrap requires jQuery (via Bootstrap). There is a project which replaces jQuery with vue.js. Link: https://bootstrap-vue.js.org/

**In conclusion**
jQuery is fine for simple sites but more advanced sites benefit from a reactive framework. vue.js is the least disruptive of the JS frameworks because our template files used in understrap can remain as they are and we can also take advantage of the existing BS vue.js fork,

**Not sure about jQuery vs vue.js**
Do search for more on jquery vs vue.js as it's quite interesting to see how they are different. Happy to discuss more if required. 👍

**Comment by Thomas-A-Reinert** [comment]
@holger1411 ?

**Comment by 0dp** [comment]
I got to ask, why would you even use WordPress for the frontend when also using a JS framework for the frontend. I mean, you are already consuming WordPress' content via the rest API. So does a Vue/react/angular WordPress theme then really make sense?

**Comment by raisonon** [comment]
Good point @0dp and there are some big advantages to using WordPress 'headerless'; separate the front and back and just use the WP Rest api. 

The main advantage to just swap out jQuery with vue.js is going to be speed. Speed to build. Speed to debug. Time saved learning the in's and out's of a new router and suchlike.

I have projects I'm working on that don't justify me to go headerless but would benefit from reactive JS like vue.js.

It means I can build using the same process I've built with many times and that I know well. I can build quickly. I just have the added benefit of using reactive JS instead of jQuery.

It's also a stepping stone to larger headerless projects.

So, you are right in one respect, but as a compromise between quality and speed I think that having vue.js instead of jQuery makes sense.

What do you think? Thanks!

**Comment by davidshq** [comment]
FYI, Gutenberg is being built using React...I'd expect to see more adaptation of React in the future.

I have heard about Vue.js being nice as a tool for progressively replacing functionality without needing to throw out everything (or a lot more) that React requires...but don't know enough about it to really say either way...still, I'd think we'd want a really robust discussion before a move to any JS framework besides jQuery...

**Comment by Thomas-A-Reinert** [comment]
Closing it since you could add Vue.js without any problem to your own theme/fork. Did this already quite a few times now.
